scio-core/src/main/scala/com/spotify/scio/values/ApproxFilter.scala

```diff
@@ -0,0 +1,53 @@
+package com.spotify.scio.values
+import com.spotify.scio.annotations.experimental
+import com.spotify.scio.coders.Coder
+
+/**
+ * A read only container for collections which can be used to check approximate
+ * membership queries.
+ *
+ * Constructors for [[ApproxFilter]] are defined using [[ApproxFilterBuilder]]
+ */
+trait ApproxFilter[-T] extends (T => Boolean) {
+
+  override def apply(t: T): Boolean = mayBeContains(t)
+
+  /**
+   * Check if the filter may contain a given element.
+   */
+  def mayBeContains(t: T): Boolean
+
+  /**
```

Comment:
Should we add a few more `val`s to reflect filter property? Assuming this abstracts an immutable, read only filter, I'm thinking:

- `numItems`: number of items inserted
- `fpp`: false positive rate at current load (this will be different from the Algebird one which tries to inefficiently compute per element `fpp`)
- `byteSize`: size of the serialized byte array, could be useful for hinting side-input or save methods

Reply:
I am not sure if we should enforce these at an Abstract level, as it might have different meanings and guarantees for different filters (BloomF/ CuckooF / QuotientF / etc). What do you think?

the `byteSize` or `estimatedByteSize` would be very useful for Side Inputs, I'll add that.

Reply:
`estimatedByteSize` sounds better than `byteSize`? I mean I know for a fact for BF, CF it's deterministic, not sure about QF. But again it depends on whether there's compression or not.

`numItems` & `fpp` are still universal across any approx filter? `numItems` may have different semantics depending on deduplication, what about `numInsertions`? That should be universal regardless of dups. Since this is an approximate filter, `fpp` should still be universal and constant? Even `fnp`, which should be `0` for most cases but we probably don't care about that assuming most impls we have guarantee 0 `fnp`.

Reply:
I think we can do away with `estimated` and keep it `sizeInBytes`. I realize that this is immutable and if there is compression, that shouldn't change.